### DISCUSSION TITLE
Signalling Server 502 when using caddy

### DISCUSSION THREAD

**Discussion by Rsclub22** [nextcloud/all-in-one#780] [Questions] [answered]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. Use Caddy when starting the container as reverse-proxy
2. Open the Firewall-Ports
3. Open Nextcloud Talk

### Expected behavior 
Everything should behave and there schould be no HTTP-Error Code
### Actual behavior 
Everything except the backend works, which repsonds with HTTP-Error Code 502 when using CADDY

### Host OS 
Fedora Linux 35 (Server Edition) x86_64

#### Nextcloud AIO version 
Fedora Linux 35 (Server Edition) x86_64
#### Current channel 
latest
#### Other valuable info 
image1 has the error message in the admin panel and image2 is the error message on a regular chat
![Image1](https://user-images.githubusercontent.com/37273508/171959170-b7898fc7-696e-4434-bb41-8f888c984df2.png)
![image2](https://user-images.githubusercontent.com/37273508/171959179-58abf112-2de3-4360-a80b-ac2f2197c39a.png)

**Comment by szaimen** [comment, marked as answer]
Did you open port 3478 UDP and TCP?

**Reply by Rsclub22** [reply]
Yes, and as yyou can see in image1 it works

**Reply by Rsclub22** [reply]
But not the Signalling

**Reply by szaimen** [reply]
Can you try disabling the talk container and enabling it again?

Also can you post your caddyfile here?

**Reply by Rsclub22** [reply]
My Caddyfile looks the following
```
nextcloud.example.com {
        header Strict-Transport-Security max-age=31536000;
        reverse_proxy 192.168.8.111:11000
}

```

**Reply by Rsclub22** [reply]
after disabling and eneabling it again, it still does not work.

**Reply by szaimen** [reply]
Can you try to stop all containers feom the aio interface, run `sudo docker rm nextcloud-aio-talk`, run `sudo docker image prune -a` and then try to start the containers again?

**Reply by Rsclub22** [reply]
still not working, maybe it is something in the talk container, i found that int the container:

```
[FATAL] [transports/janus_http.c:janus_http_init:795] Couldn't start webserver on port 8088...
[ERR] [transports/janus_nanomsg.c:janus_nanomsg_init:224] Error binding Janus API Nanomsg socket to address 'ipc:///tmp/janus.ipc': 98 (Address already in use)
[ERR] [transports/janus_websockets.c:janus_websockets_init:600] Error creating libwebsockets context...
[WARN] The 'janus.transport.websockets' plugin could not be initialized
[FATAL] [janus.c:main:5447] No Janus API transport is available... enable at least one and restart Janus
Error launching Janus (error code 1), check the logs for more details
2022-06-03 22:32:46.936657 I | main.go:200: Could not initialize janus MCU (dial tcp 127.0.0.1:8188: connect: connection refused) will retry in 16s
```

**Reply by szaimen** [reply]
I doubt that it is something with the container image itself because it would then have the same problem for everyone...

**Reply by Rsclub22** [reply]
I think that it may have to do with the caddyfile.

**Reply by Rsclub22** [reply]
Hello again, did you know which issue or discussion that was, where someone mentioned the same issue with nginx, i think he than improved the config, but i am unable to find that again. I also think it may or may not has to do something with the host, because, OnlyOffice is not working on this Server either, and no it is not because of the os, it has to do something with the Motherboard or something else. Either way, i hope it will work, with the then improved config, and if there are no improvements, i have to dig deeper, why and why not this issue exists.

**Reply by szaimen** [reply]
I know of someone that had also problems with a container image and decided to set up its raspberry pi again. Afterwards it worked...

Sorry that I have no better idea for you...
